File: 04 Kontrollstrukturen/Go/Aufgabe7.go
package main

import "fmt"

var (
	zahl1 int
	zahl2 int
)

func main() {
	fmt.Print("Bitte gib die erste Zahl ein: ")
	fmt.Scan(&zahl1)

	fmt.Print("Bitte gib die zweite Zahl ein: ")
	fmt.Scan(&zahl2)

	if (zahl1 < 1) || (zahl2 < 1) {
		fmt.Println("Beide Zahlen müssen positiv sein!")
	} else {
		fmt.Print("Der ggT von ", zahl1, " und ", zahl2, " ist ")

		for zahl1 != zahl2 {
			if zahl1 > zahl2 {
				zahl1 = zahl1 - zahl2
			} else {
				zahl2 = zahl2 - zahl1
			}
		}

		fmt.Println(zahl1, ".")
	}
}

File: 04 Kontrollstrukturen/Pascal/Aufgabe7.pas
Program Aufgabe7;

{$codepage utf8}

Var Zahl1: Integer;
    Zahl2: Integer;

Begin
    Write('Bitte gib die erste Zahl ein: ');
    Read(Zahl1);

    Write('Bitte gib die zweite Zahl ein: ');
    Read(Zahl2);

    If (Zahl1 < 1) or (Zahl2 < 1) Then
        WriteLn('Beide Zahlen müssen positiv sein!')
    Else Begin
        Write('Der ggT von ', Zahl1, ' und ', Zahl2, ' ist ');

        While Zahl1 <> Zahl2 Do
            If Zahl1 > Zahl2 Then
                Zahl1 := Zahl1 - Zahl2
            Else
                Zahl2 := Zahl2 - Zahl1;

        WriteLn(Zahl1, '.');
    End;
End.